### PR TITLE
Add ODBC extra for the production image

### PR DESCRIPTION
**Pull request by potiuk**
The ODBC extra has been missing from #18382. This PR adds the
missing extra and verifies if pyodbc is importable in the PROD
image.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -34,7 +34,7 @@
 #                        much smaller.
 #
 ARG AIRFLOW_VERSION="2.2.0.dev0"
-ARG AIRFLOW_EXTRAS="async,amazon,celery,cncf.kubernetes,docker,dask,elasticsearch,ftp,grpc,hashicorp,http,ldap,google,google_auth,microsoft.azure,mysql,pandas,postgres,redis,sendgrid,sftp,slack,ssh,statsd,virtualenv"
+ARG AIRFLOW_EXTRAS="async,amazon,celery,cncf.kubernetes,docker,dask,elasticsearch,ftp,grpc,hashicorp,http,ldap,google,google_auth,microsoft.azure,mysql,odbc,pandas,postgres,redis,sendgrid,sftp,slack,ssh,statsd,virtualenv"
```
Comment (mik-laj) [inline Dockerfile]:
```suggestion
ARG AIRFLOW_EXTRAS="amazon,async,celery,cncf.kubernetes,dask,docker,elasticsearch,ftp,google,google_auth,grpc,hashicorp,http,ldap,microsoft.azure,mysql,odbc,pandas,postgres,redis,sendgrid,sftp,slack,ssh,statsd,virtualenv"
```
Should it be sorted?

**Review by mik-laj — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (potiuk) [reply Dockerfile]: Fixed it and added pre-commit.

**Review by ashb — COMMENTED** [review]
Why do we need to include the extra in to first place when we already provide detailed instructions for a user on how to add extras?

Oh mssql support

File: scripts/ci/pre_commit/pre_commit_check_order_dockerfile_extras.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+from rich import print
+
+errors = []
+
+MY_DIR_PATH = os.path.dirname(__file__)
+SOURCE_DIR_PATH = os.path.abspath(os.path.join(MY_DIR_PATH, os.pardir, os.pardir, os.pardir))
+
+
+def _check_list_sorted(the_list: List[str], message: str) -> None:
```
Comment (mik-laj) [inline scripts/ci/pre_commit/pre_commit_check_order_dockerfile_extras.py]: Do you consider to use [`difflib`](https://docs.python.org/3/library/difflib.html)?

Comment (potiuk) [reply scripts/ci/pre_commit/pre_commit_check_order_dockerfile_extras.py]: Prefer to keep in sync with other checks. But maybe rewriting them all to difflib is a good idea to do @mik-laj  :)

Comment (mik-laj) [reply scripts/ci/pre_commit/pre_commit_check_order_dockerfile_extras.py]:
```
>>> a = "async,amazon,celery,cncf.kubernetes,docker,dask,elasticsearch,ftp,grpc,hashicorp,http,ldap,google,google_auth,microsoft.azure,mysql,pandas,postgres,redis,sendgrid,sftp,slack,ssh,statsd,virtualenv".split(",")
>>> b = "amazon,async,celery,cncf.kubernetes,dask,docker,elasticsearch,ftp,google,google_auth,grpc,hashicorp,http,ldap,microsoft.azure,mysql,odbc,pandas,postgres,redis,sendgrid,sftp,slack,ssh,statsd,virtualenv".split(",")
>>> print("\n".join(difflib.ndiff(a, b)))
+ amazon
  async
- amazon
  celery
  cncf.kubernetes
+ dask
  docker
- dask
  elasticsearch
  ftp
+ google
+ google_auth
  grpc
  hashicorp
  http
  ldap
- google
- google_auth
  microsoft.azure
  mysql
+ odbc
  pandas
  postgres
  redis
  sendgrid
  sftp
  slack
  ssh
  statsd
  virtualenv
```

Comment (potiuk) [reply scripts/ci/pre_commit/pre_commit_check_order_dockerfile_extras.py]: Feel free as a follow-up (for all pre-commits)

**Comment by potiuk** [comment]
Added extras to the docs :)

File: scripts/ci/pre_commit/pre_commit_check_order_dockerfile_extras.py (hunk 2 above)
Comment (mik-laj) [reply scripts/ci/pre_commit/pre_commit_check_order_dockerfile_extras.py]: Ok. Will do.

**Review by mik-laj — APPROVED** [review]

**Pull request merged**